File: go/dataMapTest.go
package main


func main(){
	test3()
}

func test(){
	a := map[string]string{"A":"apple", "B":"banana"}
	
	for k,v := range a{
		println(k,v)
	}
}

func test2(){
	var a = make(map[string]string)
	a["A"]="apple"
	a["B"]="banana"
	a["O"]="orange"
	delete(a, "O")

	for k,v := range a{
		println(k,v)
	}
}

func test3(){
	a := map[string]string{"A":"apple", "B":"banana"}
	
	if v, ok := a["A"]; ok {
		println(v)
	}else{
		println("orange")
	}
}

File: typescript/dataMapTest.ts

function test(){
	var a = new Map<String,String>()
	a.set("A", "apple")
	a.set("B", "banana")
	console.log(a.get("A"), a.get("B"))
}

function test2(){
	var a = new Map<String,String>()
	a.set("A", "apple")
	a.set("B", "banana")
	for(let [k, v] of a){
		console.log(k, v)
	}
	for(let k of a.keys()){
		console.log(k)
	}
	for(let k of a.values()){
		console.log(k)
	}
}

function test3(){
	var a = new Map<String,String>()
	a.set("A", "apple")
	a.set("B", "banana")
	console.log(a.size)
	a.delete("A")
	console.log(a.has("A"), a.get("A"))
}

test3()